registry.go

```diff
@@ -166,22 +180,30 @@ func (r *Registry) sendBatch(measurements []Measurement) {
 }
 
 func (r *Registry) publish() {
-	if len(r.config.Uri) == 0 {
-		return
-	}
-
-	measurements := r.Measurements()
-	r.config.Log.Debugf("Got %d measurements", len(measurements))
-	if !r.config.IsEnabled() {
-		return
-	}
+	if r.config.Uri == "" {
+		// internal publish
+		// 	We merge new metrics into the Export as sample instances are not the same for all metrics
+		new := Convert(r)
+		export := r.GetExport()
+		for k, v := range new {
+			export[k] = v
+		}
+		r.SetExport(export)
+	} else {
+		// external publish
+		measurements := r.Measurements()
```

Comment:
Small nit, nothing blocking on this PR - In Go, it's more idiomatic to avoid using `else`, prefer pushing things left and return early. Further, since each of these blocks is doing something different, it might be best to break them out into separate methods, reducing the body of this method.